src/client/public/scripts/components/row-header.ts

```diff
@@ -0,0 +1,102 @@
+import {html} from '../../../../../node_modules/lit-html/lib/lit-extended.js';
+import {TemplateResult} from '../../../../../node_modules/lit-html/lit-html.js';
+import {timeToString} from '../dash/utils/time-to-string.js';
+
+import {BaseElement} from './base-element.js';
+import {DashboardRowData} from './dashboard-row.js';
+
+export class RowHeader extends BaseElement {
+  private _rowData: DashboardRowData|undefined = undefined;
+  private _extraHeaderData: TemplateResult[]|undefined = undefined;
+
+  constructor() {
+    super();
+  }
+
+  get rowData(): DashboardRowData|undefined {
+    return this._rowData;
+  }
+
+  set rowData(data: DashboardRowData|undefined) {
+    this._rowData = data;
+    this.requestRender();
+  }
+
+  get extraHeaderData() {
+    return this._extraHeaderData;
+  }
+
+  set extraHeaderData(data: TemplateResult[]|undefined) {
+    this._extraHeaderData = data;
+    this.requestRender();
+  }
+
+  render() {
+    async function handleRowClick(event: Event) {
```

Comment:
This is going to force a mixture of logic with view, is that normal in WC?

Reply:
Yes, but also this part is identical to before.